packages/webhid/src/methods.ts

```
@@ -43,7 +57,12 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
 	)
 
 	// Wait for the device to initialize:
-	await xkeys.init()
+	try {
+		await xkeys.init()
 
-	return xkeys
+		return xkeys
+	} catch (e) {
+		await deviceWrap.close()
+		throw new Error('Failed to initialise device')
```

We should throw the original error here